### ISSUE TITLE
Implement audioMasterProcessEvents

### ISSUE THREAD

**Issue by robbert-vdh** [robbert-vdh/yabridge#5]
This would allow plugins to send MIDI data back to the host. This is what plugins like Cthulhu use to generate MIDI data such as notes or CC.

**Comment by robbert-vdh** [comment]
@cameronleger 
I've finally had a bit of time to actually add this. Setting up builds for Ubuntu 18.04 took an unexpectedly large amount of time. I only tested this using Cthulhu, but there should not be any issues for other plugins. Since you're running Arch, if you'd like to give this a shot you can install `yabridge-git` from the AUR. You will need to downgrade to wine-staging 5.6 first since winelib builds with Wine 5.7 are broken.

**Comment by cameronleger** [comment]
Thanks for the git package!

Now, I'm all the way back on wine-staging 4.20 because I rarely update that; it's the most common cause of issues and frequently plugins will require re-activation after a wine update.

Also, I haven't updated in a week.

So, the build did not succeed, lol. I'll try updating some things and get back to this. The build log is below in case you're feeling curious.

```
[20/20] Linking target libyabridge.so
FAILED: libyabridge.so 
c++  -o libyabridge.so 'yabridge@sha/src_common_logging.cpp.o' 'yabridge@sha/src_common_serialization.cpp.o' 'yabridge@sha/src_plugin_host-bridge.cpp.o' 'yabridge@sha/src_plugin_plugin.cpp.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libyabridge.so -ldl /usr/lib32/libboost_filesystem.a -Wl,--end-group -pthread
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(operations.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(path.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(exception.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(directory.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) [clone .isra.0]':
(.text.unlikely+0x4f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) [clone .isra.0]':
(.text.unlikely+0x4f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::filesystem::path const&, boost::system::error_code) [clone .cold]':
(.text.unlikely+0x1e8): undefined reference to `operator delete(void*, unsigned int)'
...
...
many more errors like:
undefined reference to `operator delete(void*, unsigned int)'
```

**Comment by cameronleger** [comment]
This may be more of a giveaway for the issue; after an update the same thing happens. Will try updating wine-staging next :X

```
==> Starting build()...
The Meson build system
Version: 0.54.1
Source dir: /tmp/makepkg/yabridge-git/src/yabridge
Build dir: /tmp/makepkg/yabridge-git/src/yabridge/build
Build type: cross build
WARNING: Unknown options: "build.cpp_std"
The value of new options can be set with:
meson setup <builddir> --reconfigure -Dnew_option=new_value ...
```

**Comment by robbert-vdh** [comment]
Do you by any chance have an old version of `lib32-boost-libs` installed? The package had not been maintained for a few years, and I adopted it and updated it for Boost 1.72 a week ago. Otherwise I'm not quite sure what would be causing these issues since I've built my copy of lib32-boost-libs with the exact same PKGBUILD.

**Comment by cameronleger** [comment]
Probably not. I assume this was updated during the latest build I tried, but I have this:
```
aur/lib32-boost-libs 1.72.0-1 (lib32) [installed]

The Meson build system
Version: 0.54.1
Source dir: /tmp/makepkg/yabridge-git/src/yabridge
Build dir: /tmp/makepkg/yabridge-git/src/yabridge/build
Build type: cross build
WARNING: Unknown options: "build.cpp_std"
The value of new options can be set with:
meson setup <builddir> --reconfigure -Dnew_option=new_value ...
Project name: yabridge
Project version: 1.0.0
C++ compiler for the build machine: c++ (gcc 9.3.0 "c++ (Arch Linux 9.3.0-1) 9.3.0")
C++ linker for the build machine: c++ ld.bfd 2.34
000b:fixme:winediag:start_process Wine Staging 4.20 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
C++ compiler for the host machine: wineg++ (gcc 9.3.0 "g++ (Arch Linux 9.3.0-1) 9.3.0")
C++ linker for the host machine: wineg++ ld.bfd 2.34
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Configuring config.h using configuration
Run-time dependency Boost (found: filesystem) found: YES 1.72.0 (/usr)
```

Still waiting on the wine-staging-git to compile against 5.6 before trying again.

**Comment by cameronleger** [comment]
Well, the lib32 must not be it, since without that dependency and with bit-bridge set to false, it still fails in the same way. Also, I built wine-staging-git against 5.6 and the same issue still occurs.

**Comment by robbert-vdh** [comment]
That Meson output is completely normal (or at least the part until the last line you pasted, with `Run-time dependency Boost (found: filesystem) found: YES 1.72.0 (/usr)` is, is some part of the output missing?). There should not be an issue with older Wine versions if you're compiling yourself.

Now that I'm actually reading it better, it looks like those linking errors are with `libyabridge.so` failing to link to 32-bit Boost. That should not be happening, since the library is built as 64-bit only. Since the boost package is a dependency of `yabridge-git`, the 64-bit version at `/usr/lib/libboost_filesystem.a` does exist, right? And another thing to check (would be strange, but still), all of the following environment variables should be empty while you're first running `meson setup`: `CC`, `CXX`, `CFLAGS`, `CXXFLAGS` and `LDFLAGS`.

**Comment by cameronleger** [comment]
Good points. I thought it was funny that it was trying against the 32-bit version even though I flipped the bit-bridge option to false. None of those flags are present in my environment. Here's the full log output from the latest failed build; I do this from the PKGBUILD while removing the lib32-boost dependency, swapping the bit-bridge flag to false, and adding `-v` to the build statement.

```
The Meson build system
Version: 0.54.1
Source dir: /tmp/makepkg/yabridge-git/src/yabridge
Build dir: /tmp/makepkg/yabridge-git/src/yabridge/build
Build type: cross build
WARNING: Unknown options: "build.cpp_std"
The value of new options can be set with:
meson setup <builddir> --reconfigure -Dnew_option=new_value ...
Project name: yabridge
Project version: 1.0.0
C++ compiler for the build machine: c++ (gcc 9.3.0 "c++ (Arch Linux 9.3.0-1) 9.3.0")
C++ linker for the build machine: c++ ld.bfd 2.34
000b:fixme:winediag:__wine_start_process Wine Staging 5.6 is a testing version containing experimental patches.
000b:fixme:winediag:__wine_start_process Please mention your exact version when filing bug reports on winehq.org.
002a:fixme:ntdll:server_ioctl_file Unsupported ioctl 24000 (device=2 access=1 func=0 method=0)
002a:fixme:ntdll:server_ioctl_file Unsupported ioctl 24000 (device=2 access=1 func=0 method=0)
002a:fixme:ntdll:server_ioctl_file Unsupported ioctl 24000 (device=2 access=1 func=0 method=0)
002a:fixme:ntdll:server_ioctl_file Unsupported ioctl 24000 (device=2 access=1 func=0 method=0)
002a:fixme:ntdll:server_ioctl_file Unsupported ioctl 24000 (device=2 access=1 func=0 method=0)
C++ compiler for the host machine: wineg++ (gcc 9.3.0 "g++ (Arch Linux 9.3.0-1) 9.3.0")
C++ linker for the host machine: wineg++ ld.bfd 2.34
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Configuring config.h using configuration
Run-time dependency Boost (found: filesystem) found: YES 1.72.0 (/usr)
Downloading bitsery source from https://github.com/fraillt/bitsery/archive/d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e.zip
Downloading file of unknown size.
Downloading bitsery patch from file:./subprojects/bitsery-patch.tar.xz
Downloading: 100%|████████████████████████████████████████| 340/340 [00:00<00:00, 3106891.85bytes/s]

|Executing subproject bitsery method meson 
|
|Project name: bitsery
|Project version: 5.0.3
|C++ compiler for the build machine: c++ (gcc 9.3.0 "c++ (Arch Linux 9.3.0-1) 9.3.0")
|C++ linker for the build machine: c++ ld.bfd 2.34
|C++ compiler for the host machine: wineg++ (gcc 9.3.0 "g++ (Arch Linux 9.3.0-1) 9.3.0")
|C++ linker for the host machine: wineg++ ld.bfd 2.34
|Build targets in project: 1
|Subproject bitsery finished.

Run-time dependency threads found: YES
Found pkg-config: /usr/bin/pkg-config (1.6.3)
Run-time dependency xcb found: YES 1.14
Build targets in project: 3

yabridge 1.0.0

  Subprojects
    bitsery: YES

Found ninja-1.10.0 at /usr/bin/ninja
ninja: Entering directory `build'                                                                   
[1/13] /usr/bin/meson --internal vcstagger ../src/common/config/version.h.in src/common/config/version.h 1.0.0 /tmp/makepkg/yabridge-git/src/yabridge/src/common/config @VCS_VERSION@ '(.*)' /usr/bin/git describe --always
[2/13] wineg++ -Iyabridge-host@exe -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -m64 -MD -MQ 'yabridge-host@exe/src_wine-host_utils.cpp.o' -MF 'yabridge-host@exe/src_wine-host_utils.cpp.o.d' -o 'yabridge-host@exe/src_wine-host_utils.cpp.o' -c ../src/wine-host/utils.cpp
[3/13] c++ -Iyabridge@sha -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -fPIC -pthread -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -MD -MQ 'yabridge@sha/src_common_serialization.cpp.o' -MF 'yabridge@sha/src_common_serialization.cpp.o.d' -o 'yabridge@sha/src_common_serialization.cpp.o' -c ../src/common/serialization.cpp
[4/13] wineg++ -Iyabridge-host@exe -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -m64 -MD -MQ 'yabridge-host@exe/src_wine-host_editor.cpp.o' -MF 'yabridge-host@exe/src_wine-host_editor.cpp.o.d' -o 'yabridge-host@exe/src_wine-host_editor.cpp.o' -c ../src/wine-host/editor.cpp
[5/13] wineg++ -Iyabridge-host@exe -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -m64 -MD -MQ 'yabridge-host@exe/src_common_serialization.cpp.o' -MF 'yabridge-host@exe/src_common_serialization.cpp.o.d' -o 'yabridge-host@exe/src_common_serialization.cpp.o' -c ../src/common/serialization.cpp
[6/13] wineg++ -Iyabridge-host@exe -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -m64 -MD -MQ 'yabridge-host@exe/src_wine-host_vst-host.cpp.o' -MF 'yabridge-host@exe/src_wine-host_vst-host.cpp.o.d' -o 'yabridge-host@exe/src_wine-host_vst-host.cpp.o' -c ../src/wine-host/vst-host.cpp
[7/13] c++ -Iyabridge@sha -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -fPIC -pthread -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -MD -MQ 'yabridge@sha/src_common_logging.cpp.o' -MF 'yabridge@sha/src_common_logging.cpp.o.d' -o 'yabridge@sha/src_common_logging.cpp.o' -c ../src/common/logging.cpp
[8/13] c++ -Iyabridge@sha -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -fPIC -pthread -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -MD -MQ 'yabridge@sha/src_plugin_plugin.cpp.o' -MF 'yabridge@sha/src_plugin_plugin.cpp.o.d' -o 'yabridge@sha/src_plugin_plugin.cpp.o' -c ../src/plugin/plugin.cpp
[9/13] wineg++ -Iyabridge-host@exe -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -m64 -MD -MQ 'yabridge-host@exe/src_common_logging.cpp.o' -MF 'yabridge-host@exe/src_common_logging.cpp.o.d' -o 'yabridge-host@exe/src_common_logging.cpp.o' -c ../src/common/logging.cpp
[10/13] wineg++ -Iyabridge-host@exe -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -m64 -MD -MQ 'yabridge-host@exe/src_wine-host_plugin-bridge.cpp.o' -MF 'yabridge-host@exe/src_wine-host_plugin-bridge.cpp.o.d' -o 'yabridge-host@exe/src_wine-host_plugin-bridge.cpp.o' -c ../src/wine-host/plugin-bridge.cpp
[11/13] wineg++  -o yabridge-host 'yabridge-host@exe/src_common_logging.cpp.o' 'yabridge-host@exe/src_common_serialization.cpp.o' 'yabridge-host@exe/src_wine-host_editor.cpp.o' 'yabridge-host@exe/src_wine-host_plugin-bridge.cpp.o' 'yabridge-host@exe/src_wine-host_vst-host.cpp.o' 'yabridge-host@exe/src_wine-host_utils.cpp.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -mwindows -m64 -Wl,--start-group /usr/lib32/libboost_filesystem.a -lpthread /usr/lib/libxcb.so -Wl,--end-group
[12/13] c++ -Iyabridge@sha -I. -I.. -I../src/include -I../subprojects/bitsery-d24dfe14f5a756c0f8ad3d56ae6949ecc2c99b2e/include -Isrc/common/config -I/usr/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++17 -O3 -fPIC -pthread -DBOOST_FILESYSTEM_STATIC_LINK=1 -DBOOST_ALL_NO_LIB -MD -MQ 'yabridge@sha/src_plugin_host-bridge.cpp.o' -MF 'yabridge@sha/src_plugin_host-bridge.cpp.o.d' -o 'yabridge@sha/src_plugin_host-bridge.cpp.o' -c ../src/plugin/host-bridge.cpp
[13/13] c++  -o libyabridge.so 'yabridge@sha/src_common_logging.cpp.o' 'yabridge@sha/src_common_serialization.cpp.o' 'yabridge@sha/src_plugin_host-bridge.cpp.o' 'yabridge@sha/src_plugin_plugin.cpp.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libyabridge.so -ldl /usr/lib32/libboost_filesystem.a -Wl,--end-group -pthread
FAILED: libyabridge.so 
c++  -o libyabridge.so 'yabridge@sha/src_common_logging.cpp.o' 'yabridge@sha/src_common_serialization.cpp.o' 'yabridge@sha/src_plugin_host-bridge.cpp.o' 'yabridge@sha/src_plugin_plugin.cpp.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libyabridge.so -ldl /usr/lib32/libboost_filesystem.a -Wl,--end-group -pthread
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(operations.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(path.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(exception.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: i386 architecture of input file `/usr/lib32/libboost_filesystem.a(directory.o)' is incompatible with i386:x86-64 output
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) [clone .isra.0]':
(.text.unlikely+0x4f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) [clone .isra.0]':
(.text.unlikely+0x4f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::filesystem::path const&, boost::system::error_code) [clone .cold]':
(.text.unlikely+0x1e8): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::filesystem::path const&, boost::filesystem::path const&, boost::system::error_code) [clone .cold]':
(.text.unlikely+0x33b): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) [clone .isra.0]':
(.text.unlikely+0x4f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::recursive_directory_iterator_construct(boost::filesystem::recursive_directory_iterator&, boost::filesystem::path const&, unsigned int, boost::system::error_code*) [clone .cold]':
(.text.unlikely+0x596): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text.unlikely+0x606): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::detail::copy_option, boost::system::error_code*)':
(.text+0x23d): undefined reference to `operator new[](unsigned int)'
/usr/bin/ld: (.text+0x47d): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x4d6): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::current_path(boost::system::error_code*)':
(.text+0xac5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0xb7d): undefined reference to `operator new[](unsigned int)'
/usr/bin/ld: (.text+0xbc0): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0xc2e): undefined reference to `operator new[](unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::absolute(boost::filesystem::path const&, boost::filesystem::path const&)':
(.text+0x130a): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x1360): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x13d8): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x153b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x161a): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o):(.text+0x1658): more undefined references to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::read_symlink(boost::filesystem::path const&, boost::system::error_code*)':
(.text+0x23a7): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x2466): undefined reference to `operator new[](unsigned int)'
/usr/bin/ld: (.text+0x2509): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x2596): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::canonical(boost::filesystem::path const&, boost::filesystem::path const&, boost::system::error_code*)':
(.text+0x40e6): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x41b5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::temp_directory_path(boost::system::error_code*)':
(.text+0x441c): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::system_complete(boost::filesystem::path const&, boost::system::error_code*)':
(.text+0x4604): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o):(.text+0x4c30): more undefined references to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o): in function `boost::filesystem::detail::(anonymous namespace)::remove_all_aux(boost::filesystem::path const&, boost::filesystem::file_type, boost::system::error_code*)':
(.text+0x58c1): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x599a): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x59f2): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x5a4a): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x5aca): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(operations.o):(.text+0x5b22): more undefined references to `operator delete(void*, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `(anonymous namespace)::root_directory_start(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int)':
(.text+0xa3): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::operator/=(boost::filesystem::path const&)':
(.text+0x11b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x207): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x26a): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x2e5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x34b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::operator/=(char const*)':
(.text+0x41f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x50b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x55e): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x5b5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x64b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::m_append_separator_if_needed()':
(.text+0x741): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::m_erase_redundant_separator(unsigned int)':
(.text+0x7a1): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_erase(unsigned int, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::remove_filename()':
(.text+0x808): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_last_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::root_directory() const':
(.text+0xa59): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0xab4): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::m_parent_path_end() const':
(.text+0xb65): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_last_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::parent_path() const':
(.text+0xccd): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_last_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: (.text+0xdee): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0xe96): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::filename() const':
(.text+0x1034): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_last_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: (.text+0x1166): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x1298): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x12d2): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::begin() const':
(.text+0x159e): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x15f2): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::relative_path() const':
(.text+0x17dd): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: (.text+0x1962): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x1a1d): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace_aux(unsigned int, unsigned int, unsigned int, char)'
/usr/bin/ld: (.text+0x1b46): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x1bfc): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::root_name() const':
(.text+0x1de2): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::root_path() const':
(.text+0x1f52): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x1fe4): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x205f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x2102): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::m_path_iterator_increment(boost::filesystem::path::iterator&)':
(.text+0x227e): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: (.text+0x23a2): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x24b0): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace_aux(unsigned int, unsigned int, unsigned int, char)'
/usr/bin/ld: (.text+0x254a): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::extension() const':
(.text+0x297f): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::rfind(char, unsigned int) const'
/usr/bin/ld: (.text+0x29f2): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::stem() const':
(.text+0x2baa): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::rfind(char, unsigned int) const'
/usr/bin/ld: (.text+0x2c4e): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x2d32): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x2d76): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::replace_extension(boost::filesystem::path const&)':
(.text+0x2fb5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x3013): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::rfind(char, unsigned int) const'
/usr/bin/ld: (.text+0x30a1): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x30d6): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::lexically_relative(boost::filesystem::path const&) const':
(.text+0x3978): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x39cd): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x3a20): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x3a7d): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o):(.text+0x3ad0): more undefined references to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::m_path_iterator_decrement(boost::filesystem::path::iterator&)':
(.text+0x40c1): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_last_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: (.text+0x427a): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x42a6): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x438d): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_first_of(char const*, unsigned int, unsigned int) const'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(path.o): in function `boost::filesystem::path::lexically_normal() const':
(.text+0x45e8): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x49ed): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x4a91): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_mutate(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x4aeb): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x4bc5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::~filesystem_error()':
(.text+0x64): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0xc3): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::~filesystem_error()':
(.text+0x183): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(boost::filesystem::filesystem_error const&)':
(.text+0x289): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::operator=(boost::filesystem::filesystem_error const&)':
(.text+0x3e3): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::system::error_code)':
(.text+0x513): undefined reference to `operator new(unsigned int)'
/usr/bin/ld: (.text+0x5e3): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::filesystem::path const&, boost::system::error_code)':
(.text+0x6fb): undefined reference to `operator new(unsigned int)'
/usr/bin/ld: (.text+0x7b9): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x863): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::filesystem_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::filesystem::path const&, boost::filesystem::path const&, boost::system::error_code)':
(.text+0x9e1): undefined reference to `operator new(unsigned int)'
/usr/bin/ld: (.text+0xaf9): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0xb41): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0xc2b): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::what() const':
(.text+0xe1b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0xe64): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0xe80): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0xeb1): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0xef0): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0xf0c): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o):(.text+0xf3d): more undefined references to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `boost::filesystem::filesystem_error::what() const':
(.text+0xf85): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0xfc5): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x1035): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x1071): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x10b6): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: (.text+0x10e6): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_append(char const*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(exception.o): in function `void boost::sp_adl_block::intrusive_ptr_release<boost::filesystem::filesystem_error::impl, boost::sp_adl_block::thread_safe_counter>(boost::sp_adl_block::intrusive_ref_counter<boost::filesystem::filesystem_error::impl, boost::sp_adl_block::thread_safe_counter> const*)':
(.text._ZN5boost12sp_adl_block21intrusive_ptr_releaseINS_10filesystem16filesystem_error4implENS0_19thread_safe_counterEEEvPKNS0_21intrusive_ref_counterIT_T0_EE[_ZN5boost12sp_adl_block21intrusive_ptr_releaseINS_10filesystem16filesystem_error4implENS0_19thread_safe_counterEEEvPKNS0_21intrusive_ref_counterIT_T0_EE]+0x73): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::directory_iterator_increment(boost::filesystem::directory_iterator&, boost::system::error_code*)':
(.text+0x37b): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x6a0): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x6ea): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x793): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::(anonymous namespace)::recursive_directory_iterator_pop_on_error(boost::filesystem::detail::recur_dir_itr_imp*)':
(.text+0x96b): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x9d3): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::recursive_directory_iterator_pop(boost::filesystem::recursive_directory_iterator&, boost::system::error_code*)':
(.text+0xba2): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0xc5b): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o):(.text+0xd26): more undefined references to `operator delete(void*, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::directory_iterator_construct(boost::filesystem::directory_iterator&, boost::filesystem::path const&, unsigned int, boost::system::error_code*)':
(.text+0xf91): undefined reference to `operator new(unsigned int, std::nothrow_t const&)'
/usr/bin/ld: (.text+0x1046): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned int, unsigned int, char const*, unsigned int)'
/usr/bin/ld: (.text+0x1276): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x13c5): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x13da): undefined reference to `operator new(unsigned int)'
/usr/bin/ld: (.text+0x1491): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: (.text+0x14ea): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_create(unsigned int&, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::recursive_directory_iterator_increment(boost::filesystem::recursive_directory_iterator&, boost::system::error_code*)':
(.text+0x1aca): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x1ba2): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x1c0b): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x1d19): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x1e5a): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o):(.text+0x1e93): more undefined references to `operator delete(void*, unsigned int)' follow
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `boost::filesystem::detail::recursive_directory_iterator_construct(boost::filesystem::recursive_directory_iterator&, boost::filesystem::path const&, unsigned int, boost::system::error_code*)':
(.text+0x21eb): undefined reference to `operator new(unsigned int, std::nothrow_t const&)'
/usr/bin/ld: (.text+0x22ba): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x22d2): undefined reference to `operator new(unsigned int)'
/usr/bin/ld: (.text+0x23ae): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text+0x23e0): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `void boost::sp_adl_block::intrusive_ptr_release<boost::filesystem::detail::recur_dir_itr_imp, boost::sp_adl_block::thread_safe_counter>(boost::sp_adl_block::intrusive_ref_counter<boost::filesystem::detail::recur_dir_itr_imp, boost::sp_adl_block::thread_safe_counter> const*)':
(.text._ZN5boost12sp_adl_block21intrusive_ptr_releaseINS_10filesystem6detail17recur_dir_itr_impENS0_19thread_safe_counterEEEvPKNS0_21intrusive_ref_counterIT_T0_EE[_ZN5boost12sp_adl_block21intrusive_ptr_releaseINS_10filesystem6detail17recur_dir_itr_impENS0_19thread_safe_counterEEEvPKNS0_21intrusive_ref_counterIT_T0_EE]+0xe1): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: (.text._ZN5boost12sp_adl_block21intrusive_ptr_releaseINS_10filesystem6detail17recur_dir_itr_impENS0_19thread_safe_counterEEEvPKNS0_21intrusive_ref_counterIT_T0_EE[_ZN5boost12sp_adl_block21intrusive_ptr_releaseINS_10filesystem6detail17recur_dir_itr_impENS0_19thread_safe_counterEEEvPKNS0_21intrusive_ref_counterIT_T0_EE]+0x11c): undefined reference to `operator delete(void*, unsigned int)'
/usr/bin/ld: /usr/lib32/libboost_filesystem.a(directory.o): in function `void std::vector<boost::filesystem::directory_iterator, std::allocator<boost::filesystem::directory_iterator> >::_M_realloc_insert<boost::filesystem::directory_iterator>(__gnu_cxx::__normal_iterator<boost::filesystem::directory_iterator*, std::vector<boost::filesystem::directory_iterator, std::allocator<boost::filesystem::directory_iterator> > >, boost::filesystem::directory_iterator&&)':
(.text._ZNSt6vectorIN5boost10filesystem18directory_iteratorESaIS2_EE17_M_realloc_insertIJS2_EEEvN9__gnu_cxx17__normal_iteratorIPS2_S4_EEDpOT_[_ZNSt6vectorIN5boost10filesystem18directory_iteratorESaIS2_EE17_M_realloc_insertIJS2_EEEvN9__gnu_cxx17__normal_iteratorIPS2_S4_EEDpOT_]+0x68): undefined reference to `operator new(unsigned int)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by robbert-vdh** [comment]
That's so weird! But according to that log, `/usr/lib32/libboost_filesystem.a` does still exist. The package that should provide that file is called `lib32-boost-libs`, not `lib32-boost`. Maybe that's what went wrong? You can check which package owns that file using `pacman -Qo 
/usr/lib32/libboost_filesystem.a`.

But anyways, if you just want to test out the new support for plugins like Cthulhu and Scaler, I've just added automated builds for the master branch. You can find them by clicking on the 'Automated builds' badge near the top of the readme and then selecting a commit. That way you can at least test whether this works!

**Comment by cameronleger** [comment]
I'm sorry man I should be more specific with my words haha. I removed the dependency from the PKGBUILD but not my installation, and yes it's `lib32-boost-libs`.

After uninstalling that package, it does build correctly! I'm not sure why that's giving me so much trouble whereas it didn't with you. That said, I swapped to the latest automated build to get the 32-bit just in case.

Something is still going wrong. Scaler crashes immediately on load up, and with logging I can see it always ends here:
```
10:38:18 [Scaler_x64-05BINYio] >> dispatch() effStartProcess(index = 0, value = 0, option = 0, data = <nullptr>)
10:38:18 [Scaler_x64-05BINYio]    dispatch() :: 0
10:38:18 [Scaler_x64-05BINYio] >> dispatch() effProcessEvents(index = 0, value = 0, option = 0, data = <2 midi_events>)
10:38:18 [Scaler_x64-05BINYio]    dispatch() :: 1
```

I tried with Gatekeeper and that plugin gets past that, so I wonder if it's something specific to Scaler. It works with LinVST, so I know it's all configured correctly. It's worth noting that I don't think the prefix was correctly detected with either:

```
10:38:24 [Gatekeeper-6DGenruw] Initializing yabridge version 1.0.0-37-g006fb64
10:38:24 [Gatekeeper-6DGenruw] host:       '/usr/bin/yabridge-host.exe'
10:38:24 [Gatekeeper-6DGenruw] plugin:     '/mnt/games/PlayOnLinux/wineprefix/VSTsOwned/drive_c/Program Files/Steinberg/VstPlugins/Gatekeeper.dll'
10:38:24 [Gatekeeper-6DGenruw] socket:     '/tmp/yabridge-Gatekeeper-6DGenruw.sock'
10:38:24 [Gatekeeper-6DGenruw] wineprefix: '<default>'
```

In this case, `/mnt/games/PlayOnLinux/wineprefix/VSTsOwned` has a folder called `dosdevices`, so I'm not sure why that didn't work. I assume there's something with the linking that still needs to be figured out; maybe it's walking back from the folder that the files are linked into:

```
ls -lah ~/.local/share/yabridge 
Permissions Size User    Date Modified Name
lrwxrwxrwx   102 cameron  6 May 10:38  Gatekeeper.so -> /mnt/games/PlayOnLinux/wineprefix/VSTsOwned/drive_c/Program Files/Steinberg/VstPlugins/Gatekeeper.so
lrwxrwxrwx   102 cameron  6 May 10:32  Scaler_x64.so -> /mnt/games/PlayOnLinux/wineprefix/VSTsOwned/drive_c/Program Files/Steinberg/VstPlugins/Scaler_x64.so
```

**Comment by robbert-vdh** [comment]
Right now the Wine prefix detection is relative to the `.so` file, but I'll change it to be relative to the plugin's .dll file since that indeed makes more sense with this setup. I had never considered symlinks to copies of `libyabridge.so` when I designed this! I just pushed this change to the master branch.

**Comment by cameronleger** [comment]
Excellent, thank you! I tried Cthulu and it works great, whereas it crashes during any MIDI event on LinVST and Airwave. This is probably resolved!

**Comment by robbert-vdh** [comment]
This got implemented in the just released yabridge 1.1.0. Feel free to open a new issue if there are any problems with MIDI output.
